### DISCUSSION TITLE
Is there any common approach to testing Taskfile implementations?

### DISCUSSION THREAD

**Discussion by Gowiem** [go-task/task#1283] [Q&A]
At @Masterpointio, we're starting down our Task journey and we've put together [a shared repo](https://github.com/masterpointio/taskit) of Taskfiles that we'll use across our clients. I would like to add a solid test harness for those Taskfile tasks so we can start to ensure quality and that we're not going to break things. 

Is there a common community approach that those who have been using Task for a while have taken? Anything open source? 

Thanks!

**Comment by trulede** [comment]
Any E2E testing approach will work. I've developed one that uses TestScript and txtar format files. It's quite nice, simply run the testscript command with a txtar file as its argument.
